### PR TITLE
[posix-app] add log level argument

### PR DESCRIPTION
**Pull request by bukepo**

### CODE REVIEW

File: src/posix/main.c [hunk 1]
```diff
@@ -191,6 +198,7 @@ static otInstance *InitInstance(int aArgCount, char *aArgVector[])
         exit(OT_EXIT_SUCCESS);
     }
 
+    otLoggingSetLevel(logLevel);
```
Comment (jwhui) [inline src/posix/main.c]: Does this assume that `OPENTHREAD_CONFIG_ENABLE_DYNAMIC_LOG_LEVEL=1`? Is it necessary to support `OPENTHREAD_CONFIG_ENABLE_DYNAMIC_LOG_LEVEL=0`?

Comment (bukepo) [reply src/posix/main.c]: Yes, this assumes dynamic logging is enabled. It's seems we'll have to make some assumptions unless openthread-core-config.h is exported for user, which is current not exported. Thoughts?

Comment (jwhui) [reply src/posix/main.c]: If this is not optional, should we make the `DYNAMIC_LOG_LEVEL` build switch in `src/posix/Makefile-posix` mandatory?

Comment (bukepo) [reply src/posix/main.c]: `DYNAMIC_LOG_LEVEL` is set to 1 in `src/posix/Makefile-posix`. If a user overrides this to `0`, he/she can modify the main.c accordingly.

**Pull request merged**